### PR TITLE
fix permission to manage public queries not respected

### PR DESCRIPTION
**Pull request by 0xF013**
[`* `#15886` Permission to manage public queries not respected`](http://www.openproject.org/work_packages/15886)

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage. <a href='http://0xf013-feature-15886-permission-to-manage-public-queries-not-re.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

File: app/assets/javascripts/angular/work_packages/directives/options-dropdown-directive.js [hunk 1]
```diff
@@ -94,7 +94,9 @@ angular.module('openproject.workPackages.directives')
       };
 
       scope.showShareModal = function(event){
-        showExistingQueryModal.call(shareModal, event);
+        if (allowQueryAction(event, 'publicize') && allowQueryAction(event, 'star')) {
```
Comment (ulferts) [inline app/assets/javascripts/angular/work_packages/directives/options-dropdown-directive.js]: I believe this should be an `||`

That should also fix the failing cuke.

Comment (0xF013) [inline app/assets/javascripts/angular/work_packages/directives/options-dropdown-directive.js]: @ulferts ok, then I will add a check to the popup (to hide/disable the star/unstar checkbox if the start permission is missing, and the same for publicize)

**Comment by ulferts** [comment]
For a public query the following should also be inactive from the menu (and of course also not callable) if the query is a public query and the user lacks the manage_public_queries permission:

image
- Save
- Delete
- Page settings ...

File: app/assets/javascripts/angular/work_packages/directives/options-dropdown-directive.js (hunk 1 above)
Comment (ulferts) [inline app/assets/javascripts/angular/work_packages/directives/options-dropdown-directive.js]: I guess they are already somewhat disabled. At least I can not use them, it only does not look like it.

**Comment by 0xF013** [comment]
@ulferts as far I saw in code, save and delete check the _links hash and are disabled if the correct links hash is built (which I think is not the case right now). I guess it's a backend issue.

**Comment by ulferts** [comment]
@0xF013 should I take a look at it then once the front end is done?

**Comment by 0xF013** [comment]
@ulferts yes please, I'll update the task log

**Comment by ulferts** [comment]
Merging this and adding the rest as separate PR.

**Pull request merged**